lib/snapshot.js

```
@@ -0,0 +1,58 @@
+'use strict';
+const path = require('path');
+const fs = require('fs');
+const isEqual = require('lodash.isequal');
+const mkdirp = require('mkdirp');
+const globals = require('./globals');
+
+class Snapshot {
+	constructor(testPath, options) {
+		if (!testPath) {
+			throw new TypeError('Test file path is required');
+		}
+
+		this.dirPath = path.join(path.dirname(testPath), '__snapshots__');
+		this.filePath = path.join(this.dirPath, path.basename(testPath) + '.snap');
+		this.tests = {};
+		this.options = options || {};
+
+		if (fs.existsSync(this.filePath)) {
+			this.tests = JSON.parse(fs.readFileSync(this.filePath));
```

I'd just read the file and ignore any errors:

```js
let contents
try {
  contents = fs.readFileSync(this.filePath);
} catch (err) {}

this.tests = contents ? JSON.parse(contents) : {}
```